ValveKeyValue/ValveKeyValue.Test/Text/ConditionalTestCase.cs

```diff
@@ -105,6 +106,52 @@ public void ConditionalInKey()
             Assert.That((string)children[0].Value, Is.EqualTo("windows 32-bit"));
         }
 
+        [Test]
+        public void ConditionalBeforeObject()
+        {
+            var data = ParseResource("Text.conditional_before_object_value.vdf");
+            Assert.That(data, Is.Not.Null);
+            Assert.That(data.Value.ValueType, Is.EqualTo(KVValueType.Collection));
+
+            var children = data.Children.ToArray();
+            Assert.That(children, Has.Length.EqualTo(0));
+
+            data = ParseResource("Text.conditional_before_object_value.vdf", new string[] { "WIN32" });
+            Assert.That(data, Is.Not.Null);
+            Assert.That(data.Value.ValueType, Is.EqualTo(KVValueType.Collection));
+
+            children = data.Children.ToArray();
+            Assert.That(children, Has.Length.EqualTo(1));
+            Assert.That(children[0].Name, Is.EqualTo("operating system"));
+            Assert.That((string)children[0].Value, Is.EqualTo("windows 32-bit"));
+        }
+
+        [Test]
+        public void ConditionalBetweenKeyAndValue()
```

Comment:
Oops, forgot to save this comment.

Validated this against Valve's tier1 implementation:

```
> steamdb_vkv_test.exe -file conditional_between_key_and_value.vdf
Name: test case
    operating system (string): windows 32-bit
```